plugin.py

```diff
@@ -128,40 +136,49 @@ def is_valid_for_view(self, view: sublime.View) -> bool:
     def _handle_log_message_notification(self, payload: CopilotPayloadLogMessage) -> None:
         pass
 
+    @notification_handler(NTFY_PANEL_SOLUTION)
+    def _handle_panel_solution_notification(self, payload: CopilotPayloadPanelSolution) -> None:
+        panel_id = int(remove_prefix(payload.get("panelId"), "copilot://"))
+        target_view = first(all_st_views(), lambda view: view.id() == panel_id)
+        if not target_view:
+            return
+
+        preprocess_panel_completions(target_view, [payload])
+        panel_completions = get_copilot_view_setting(target_view, "panel_completions", [])
+        panel_completions.append(payload)
+
+        set_copilot_view_setting(target_view, "panel_completions", panel_completions)
+        ViewCompletionManager(target_view).update_panel_completions()
+
+    @notification_handler(NTFY_PANEL_SOLUTION_DONE)
+    def _handle_panel_solution_done_notification(self, payload) -> None:
+        panel_id = int(remove_prefix(payload.get("panelId"), "copilot://"))
+        target_view = first(all_st_views(), lambda view: view.id() == panel_id)
+        if not target_view:
+            return
+
+        set_copilot_view_setting(target_view, "is_waiting_panel_completions", False)
+
     @notification_handler(NTFY_STATUS_NOTIFICATION)
-    def _handle_status_notification(self, payload: CopilotPayloadStatusNotification) -> None:
+    def _handle_status_notification_notification(self, payload: CopilotPayloadStatusNotification) -> None:
         pass
 
     def request_get_completions(self, view: sublime.View) -> None:
         ViewCompletionManager(view).hide()
 
         session = self.weaksession()
-        syntax = view.syntax()
         sel = view.sel()
-        if not (self.get_has_signed_in() and session and syntax and len(sel) == 1):
+        if not (self.get_has_signed_in() and session and len(sel) == 1):
             return
 
-        cursor = sel[0]
```

Comment:
Just a note regarding the idea I shared above.

Instead of handling everything with utility functions(like `prepare_completion_request`) we could go OOP and have something like `Request(view).perform()`

We could introduce a base abstract class interface and require `perform` and `handle` methods to be implemented for every `BaseRequest` subclass. 

It is not critical for this PR I think, and we should probably refactor this after we merge it

Reply:
> It is not critical for this PR I think, and we should probably refactor this after we merge it

👍